app/main.py

```diff
@@ -1,3 +1,78 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: int | Distance | float) -> Distance:
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        if isinstance(other, int):
+            return Distance(self.km + other)
+        if isinstance(other, float):
+            return Distance(self.km + other)
```

Comment:
You return same Distance(self.km + other) two times, possible to combine them: if isinstance(other, int | float):... or somehow else. 
The same about other methods below.

Reply:
Thanks for the review! I didn't notice for the first time that I could simplify these expressions.